File: Leetcode/Array/#566. Reshape the Matrix.cpp
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& nums, int r, int c) {
        int n = nums.size();
        if(n==0) return nums;
        int m = nums[0].size();
        int total = n*m;
        if(total!=r*c) return nums;
        
        vector<vector<int>>res(r,vector<int>(c));
        
       for(int i=0; i<total; i++){
           res[i/c][i%c]=nums[i/m][i%m];
       }
        return res;
    }
};

File: Array/#566. Reshape the Matrix.py
'''
TC: O(mn)
SC: O(1)
'''
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        m, n = len(mat), len(mat[0])
        res = [[None for _ in range(c)] for _ in range(r)]
        if r * c != m * n:
            return mat

        for i in range(m):
            for j in range(n):
                idx = i * n + j
                res[idx // c][idx % c] = mat[i][j]

        return res